src/commands/moderators/deleteAllMessage.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  ApplicationCommandType,
+  ButtonStyle,
+  ChannelType,
+  ComponentType,
+  DiscordAPIError,
+  PermissionsBitField,
+} from 'discord.js'
+
+import { CommandHandlerConfig } from '../../types/CommandHandlerConfig.js'
+
+export default {
+  data: {
+    name: 'Prune messages',
+    type: ApplicationCommandType.Message,
+    defaultMemberPermissions: PermissionsBitField.Flags.ManageChannels,
+    dmPermission: false,
+  },
+  ephemeral: true,
+  execute: async (client, interaction) => {
+    if (!interaction.guild || !interaction.isContextMenuCommand()) return
+
+    // TODO: Confirmation banner
+    await interaction.followUp({
+      content: 'Are you sure you want to prune all messages from this user?',
+      components: [
+        {
+          type: ComponentType.ActionRow,
+          components: [
+            {
+              type: ComponentType.Button,
+              style: ButtonStyle.Primary,
+              label: 'Yes',
+              customId: 'yes',
+            },
+            {
+              type: ComponentType.Button,
+              style: ButtonStyle.Danger,
+              label: 'No',
+              customId: 'no',
+            },
+          ],
+        },
+      ],
+    })
+
+    // Await button interaction for confirmation
+    const buttonInteraction = await interaction.channel?.awaitMessageComponent({
+      filter: (i) => i.customId === 'yes' || i.customId === 'no',
+      time: 10000, // Adjust timeout as needed
+    })
```

Comment:
Will there be a problem if many people are using the command at the same time?

Also please tag user in confirmation message so admin can confirm

Reply:
hmm interesting point for race condition, maybe I can handle by stop loop message in channel when it got an error by not found message (other moderator already deleted the message)